blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/Locator.java

```diff
@@ -86,7 +86,7 @@ public boolean equals(Locator other) {
 
     public static Locator createLocatorFromPathComponents(String tenantId, String... parts) throws IllegalArgumentException {
         if(StringUtils.isEmpty(tenantId) || parts == null)
-            throw new IllegalArgumentException("'tenantId' is null or empty OR 'parts' is null.");
+            log.error("'tenantId' is null or empty OR 'parts' is null.");
```

Comment:
So now it'll create a bogus Locator. Will that just cause the query to not match anything? I prefered the exception throwing, @GeorgeJahad, since that can be ultimately caught by the web handler layer and translated into a 400 back to the client. With all that said, I don't mind either way since you two have discussed this.

Reply:
I guess my philosophy is that user input should be checked when that input first enters the system, not creating exceptions throughout the system.

As I discussed with Mritunjay, unless we are sure these exceptions are properly handled somewhere, I'm concerned that they will cause the thread to halt, eventually bringing the system to its knees.

I guess the correct solution is to just be sure we have the correct exception handlers everywhere, but I'm not convinced that's trivial, and opens up a whole new issue that I really don't want to address in this PR.

If you/Jay feel strongly, I would support opening a new PR to address those issues, but this one is so complicated already I don't want to add another set of issues to it.  What do you guys think, @itzg / @mritunjaykumar ?

Reply:
I agree with @itzg . That's why I implemented that way earlier. But, @GeorgeJahad had a valid point that what if this exception blows something. So, I removed that exception just based on that fear. This check-in is massive in itself as it's literally unrooting native ES client calls, and adding another variable to the mix was kind of debatable risk particularly in this check-in. IMHO, we can take this work, when we know for certain that it has no impact on other code. Besides, AFAIK, it will never hit this code path based on whatever I have seen so far. So, I would let it be there for the next refactor.